tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -118,27 +113,23 @@ private void Dispose(bool disposing)
             }
         }
 
-        private ClusterType MapToplogyToClusterType(BsonValue topology)
-        {
-            switch (topology.AsString)
-            {
-                case "single": return ClusterType.Standalone;
-                case "replicaset": return ClusterType.ReplicaSet;
-                case "sharded": return ClusterType.Sharded;
-                default: throw new ArgumentException("Unknown topology type");
-            }
-        }
-
         private void Run(BsonDocument shared, BsonDocument test)
         {
             if (test.Contains("skipReason"))
             {
                 throw new SkipException($"Test skipped because {test["skipReason"]}.");
             }
 
-            if (shared.TryGetValue("topology", out var topology))
+            CheckTransactionFeature();
```

Comment:
I'm starting to think we should have left the original CheckClusterTopology method and call completely alone and if the JSON file contained:

"topology" => call the original unchanged CheckClusterTopology method just like before
"runOn" => call RequireServer.Check().RunOn(...)

Reply:
I don't think so yet, as you know `RunOn` contains validation on `topology`. So, in case if we have both options, we will have unnecessary validation steps.

Reply:
The same answer as above, as soon as we will have only `runOn` we will need just simply remove mapping `topology` format on `runOn`.

So, I think the proposed ways are equivalents with the difference that the current version doesn't have double validation on `topology` and as the result `code duplication`.

Reply:
There is no real code duplication.

The code for "topology" implements support for those JSON files that still contain "topology", and the code for "runOn" implements support for those JSON files that now contain "runOn". I'm pretty sure no JSON file will contain both "topology" and "runOn", but if they do then we will just validate both of them.

Reply:
These codes are the same and work for the same purpose:

https://github.com/mongodb/mongo-csharp-driver/blob/7cd604ab4fee1c3a02fd56e7b3f99215055785f4/tests/MongoDB.Driver.Core.TestHelpers/XunitExtensions/RequireServer.cs#L246

https://github.com/mongodb/mongo-csharp-driver/blob/87a83397ed8644c46a6433f2c6946f5de61b0673/tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs#L121

So for me, it's `code duplication`.
But I don't think that we need to spend more time for this PR. I will do as you propose.